emukit/examples/simple_gp_model/simple_gp_model.py

```diff
@@ -0,0 +1,121 @@
+from typing import Tuple
+
+import numpy as np
+import scipy.linalg
+import scipy.optimize
+
+from emukit.core.interfaces import IModel
+
+
+class SimpleGaussianProcessModel(IModel):
+    """
+    This model is a Gaussian process with an RBF kernel, with no ARD. It is used to demonstrate uses of emukit,
+    it does not aim to be flexible, robust or fast.
+    """
+    def __init__(self, x: np.ndarray, y: np.ndarray):
+        """
+        :param x: (n_points, n_dims) array containing training features
+        :param y: (n_points, 1) array containing training targets
+        """
+        self.x = x
+        self.y = y
+        self.lengthscale = 1
+        self.kernel_variance = 1
+        self.likelihood_variance = 1
+        self.jitter = 1e-6
+
+    def optimize(self) -> None:
+        """
+        Optimize the three hyperparameters of the model, namely the kernel variance, kernel lengthscale and likelihood
+        variance
+        """
+        def optimize_fcn(x):
+            # take exponential to ensure positive values
+            x = np.exp(x)
+            self.lengthscale = x[0]
+            self.kernel_variance = x[1]
+            self.likelihood_variance = x[2]
+            return self._negative_marginal_log_likelihood()
+
+        scipy.optimize.minimize(optimize_fcn, np.log(np.array([self.lengthscale,
+                                                               self.kernel_variance,
+                                                               self.likelihood_variance])))
+
+    def predict(self, x_new: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Predict from model
+
+        :param x_new: (n_points, n_dims) array containing points at which the predictive distributions will be computed
+        :return: Tuple containing two (n_points, 1) arrays representing the mean and variance of the predictive
+                 distribution at the specified input locations
+        """
+        K = self._calc_kernel(self.x)
+        idxs = list(range(self.x.shape[0]))
+        K[idxs, idxs] += self.jitter + self.likelihood_variance
+        L = np.linalg.cholesky(K)
+
+        K_xs = self._calc_kernel(self.x, x_new)
+
+        tmp = scipy.linalg.solve_triangular(L, K_xs, lower=True)
+        tmp2 = scipy.linalg.solve_triangular(L, self.y, lower=True)
+
+        mu = np.dot(tmp.T, tmp2)
+        var = (self.kernel_variance - np.sum(np.square(tmp), axis=0))[:, None]
+        return mu, var
+
+    def _calc_kernel(self, X, X2=None):
```

Comment:
type annotation missing, and docstring params and return missing

Reply:
Added the docstrings now